examples/simple/main_test.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"io"
+	"strconv"
+	"testing"
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/bubbletea/teatest"
+)
+
+func TestApp(t *testing.T) {
+	t.Parallel()
+	for i := 0; i < 3; i++ {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			t.Parallel()
+			m := model(10)
+			teatest.TestModel(
+				t, m,
+				func(p teatest.Program, in io.Writer) {
+					time.Sleep(time.Second + time.Millisecond*200)
+					p.Send("ignored msg")
+					p.Send(tea.KeyMsg{
+						Type: tea.KeyEnter,
+					})
+				},
+				func(out []byte) {
+					teatest.RequireEqualOutput(t, out)
+				},
```

Comment:
Firstly, thank you for this PR! I'm currently building my first bubbles and want to test them. Here's some feedback based on this PR based on my own testing efforts in https://github.com/twpayne/chezmoi/pull/2359.

At this point, I'm testing that my models have certain properties after receiving a bunch of messages, so I'd like to include that in the testing framework. A specific suggestion would be to make `teatest.Model` accept [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) (just like most of bubbletea already does), so I can write code like:

```go
    teatest.Model(
        t, m,
        teatest.WithSender(p teatest.Program, in io.Writer) {
            // as above
        },
        teatest.WithRequireOutput(/* ... */),
        teatest.WithValidateFinalModel(func(m tea.Model) err {
            // check some properties of m
        },
    )
```

Reply:
That sounds like a good idea, yes. Will try to do something later today 🙏